core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -920,7 +986,7 @@ class Partition(val topicPartition: TopicPartition,
      * is violated, that replica is considered to be out of sync
      *
      **/
-    val candidateReplicaIds = inSyncReplicaIds - localBrokerId
+    val candidateReplicaIds = isrState.maximalIsr - localBrokerId
```

Comment:
This is related to my comment above for the ISR expansion case, but it is a bit confusing to use maximal ISR when the expectation is that we will not shrink as long as we have a pending update inflight. Would it be better to check for inflights and document that this method will return an empty set as long as there is a pending AlterIsr request?

Reply:
Makes sense, that will also satisfy your other comment about not checking for inflight requests within the write lock

Reply:
Also, yes it's confusing to refer to `maximalIsr` here even though it should always equal the committed ISR at this point (assuming we check for inflight first).